contracts/FindLeaseMarketDirectOfferSoft.cdc

```diff
@@ -44,9 +39,9 @@ pub contract FindLeaseMarketDirectOfferSoft {
 				self.offerCallback.check() : "Bidder unlinked the bid collection capability."
 				self.pointer != nil : "Please accept offer"
 			}
-			self.offerCallback.borrow()!.acceptNonEscrowed(self.getLeaseName())
 			let pointer = self.pointer as! FindLeaseMarket.AuthLeasePointer
 			pointer.move(to: self.offerCallback.address)
+			self.offerCallback.borrow()!.acceptNonEscrowed(self.getLeaseName())
```

Comment:
move the sequence so that the name is first move to the buyer, assert that the buyer gets the lease then release the fund